app.py

```diff
@@ -98,26 +97,83 @@ def find_font(name):
 videos = get_videos(searchPath)
 video_names = get_video_names(searchPath)
 
-# Preload the response; it's always the same
-subs = subs2json(load_subs(videos))
+# Loads the video subtitles
+video_subs = load_subs(videos)
+# Creates a list of video data with subtitles
+videos_with_subs = [{ 'title': video_names[idx], 'id': idx, 'subs': v[0] } for idx, v in enumerate(video_subs)]
+# Convert sub data to JSON data
+subs = subs2json(video_subs)
+# Creates a list of the installed fonts
+fonts = font_manager.findSystemFonts(fontpaths=None, fontext='ttf')
 
 # Find a suitable font
 font = find_font('Noto')
 
+# Jinja2 templates
+templates = Environment(loader = FileSystemLoader('templates'))
+
 logger.info("ready to start the application")
 
+# Serves anything stored in public folder
+@app.route("/public/<path:path>")
+def get_public(path):
+    return send_from_directory("public", path)
+
+# Shows the main UI
+@app.route("/")
+def get_index():
+    search = request.args.get("q")
+    if search is not None:
+        filtered_subs = [{ **video, 'subs': [sub for sub in video['subs'] if search.lower() in sub.text] } for video in videos_with_subs]
+        return render_template("sublist.html", videos=filtered_subs)
+    else:
+        return render_template("index.html", videos=videos_with_subs)
+    
+# Creates the GIF settings that is shown at the bottom of the page
+@app.route("/sub_form/<video_id>/<sub_id>")
```

Comment:
Similar question to an earlier one: aren't these requests pointless? All necessary information to prefill the form is known to the client (since it's part of the subs), so why send another request to the server _every_ time a subtitle is clicked?

Reply:
That's how HTMX works babey

Reply:
And also again here: caching